### ISSUE TITLE
Resource 'data.baremetal_core_vnic_attachments.master_node_vnics' does not have attribute 'vnic_attachments.0.vnic_id'

### ISSUE THREAD

**Issue by scross01** [oracle/terraform-provider-baremetal#14]
Scenario:  launching multiple instances and trying to get the public_ip_address of each instance from the vnic data source.  The following configuration fails with the error below on apply

```
variable "master_count" { default = 1 }

# Create the Master Nodes
resource "baremetal_core_instance" "master_node" {
  count = "${var.master_count}"
  availability_domain = "CxvG:PHX-AD-2"
  compartment_id = "${var.compartment_ocid}"
  display_name = "${format("master%1d", count.index + 1)}"
  image = "ocid1.image.oc1.phx.aaaaaaaaclseho77fcdfgejstt2bflkugcx5waa6bhconbokvhdp3qw7txlq"
  shape = "VM.Standard1.1"
  subnet_id = "ocid1.subnet.oc1.phx.aaaaaaaaiejsvcgvvmthx62fij6gvdlsrjmu3eczkwiusktz4axu4gel5q3q"
  metadata {
    ssh_authorized_keys = "${file("~/.ssh/id_rsa.pub")}"
  }
}

data "baremetal_core_vnic_attachments" "master_node_vnics" {
  count = "${var.master_count}"
  availability_domain = "CxvG:PHX-AD-2"
  compartment_id = "${var.compartment_ocid}"
  instance_id = "${element(baremetal_core_instance.master_node.*.id, count.index)}"
}

data "baremetal_core_vnic" "master_node_vnics" {
  count = "${var.master_count}"
  vnic_id = "${element(data.baremetal_core_vnic_attachments.master_node_vnics.*.vnic_attachments.0.vnic_id, count.index)}"
}
```

```
Error running plan: 1 error(s) occurred:

* Resource 'data.baremetal_core_vnic_attachments.master_node_vnics' does not have attribute 'vnic_attachments.*.vnic_id' for variable 'data.baremetal_core_vnic_attachments.master_node_vnics.*.vnic_attachments.0.vnic_id'
```

However, if the `data "baremetal_core_vnic" "master_node_vnics"` item is omitted on the first `terraform apply`, and then added to the config and `terraform apply` is run again - the configuration works okay and the data source `data.baremetal_core_vnic_attachments.master_node_vnics.*.public_ip_address` is available for use in additional config.

**Comment by craigcarl-oracle** [comment]
@scross01 I pushed a new binary yesterday designed to fixed this issue, are you using the newest version?

**Comment by scross01** [comment]
I tested with the latest build  (v1.0.3)

The problem only seems to occur when using the `count` parameter for launching multiple instances. I wonder if it could be a core problem with the `${element()}` function because there are two indexed items in the data item reference - although it's odd that it works okay on a second apply.

**Comment by lfeldman** [comment]
I am not sure if you have noticed but I think it works different when you try to obtain vnics info for VMs and BMs. For BMs works perfectly fine. For VMs it looks like there is no vnic at all... But I am still wondering how to posses for further purposes VM's public IP address...

**Comment by adubnik** [comment]
Having the same issue while using count. When TF apply starts it throws errors

```
* data.baremetal_core_vnic.InstanceVnic[0]: Resource 'data.baremetal_core_vnic_attachments.InstanceVnics' does not have attribute 'vnic_attachments.0.vnic_id' for variable 'data.baremetal_core_vnic_attachments.InstanceVnics.*.vnic_attachments.0.vnic_id'
* data.baremetal_core_vnic.InstanceVnic[1]: Resource 'data.baremetal_core_vnic_attachments.InstanceVnics' does not have attribute 'vnic_attachments.0.vnic_id' for variable 'data.baremetal_core_vnic_attachments.InstanceVnics.*.vnic_attachments.0.vnic_id'
```

Where code look like this

```
# Gets a list of vNIC attachments on the instance
data "baremetal_core_vnic_attachments" "InstanceVnics" { 
    count = "${var.machine_count}"
    compartment_id = "${var.compartment_ocid}" 
    availability_domain = "${lookup(data.baremetal_identity_availability_domains.ADs.availability_domains[var.AD - 1],"name")}" 
    instance_id = "${element(baremetal_core_instance.TFInstance.*.id,count.index)}" 
}  

data "baremetal_core_vnic" "InstanceVnic" { 
    depends_on = ["baremetal_core_instance.TFInstance"]
    count = "${var.machine_count}"
    vnic_id = "${data.baremetal_core_vnic_attachments.InstanceVnics.*.vnic_attachments.0.vnic_id[count.index]}"
}
```
If I comment the problem and provision configuration and then uncomment the code it works just fine. Seem TF does check ahead of time when tfstate is formed.

**Comment by codycushing** [comment]
`baremetal_core_instance...public_ip|private_ip` will be available in v1.0.10, no longer requiring explicit vnic lookup.
